addons/web_editor/static/src/js/backend/field_html.js

```diff
@@ -235,9 +235,9 @@ var FieldHtml = basic_fields.DebouncedField.extend(TranslatableFieldMixin, {
         this.$content.toggleClass('d-none');
         if ($codeview.hasClass('d-none')) {
             this.wysiwyg.odooEditor.observerActive();
-            this.wysiwyg.setValue($codeview.val());
+            this.wysiwyg.$editable.find('.o_editable')[0].innerHTML = $codeview[0].value;
         } else {
-            $codeview.val(this.$content.html());
+            $codeview.val(this.wysiwyg.$editable.find('.o_editable')[0].innerHTML);
```

Comment:
It is not clear to me. How is `this.wysiwyg.$editable.find('.o_editable')[0].innerHTML` different than `this.$content.html()` and what role does this play in fixing the traceback ?

Reply:
@dmo-odoo 
Using "this.$content.html()" will give the innerHTML of the editable zone. However, the innerHTML also includes the droppable zone. As a result, when the codeview is emptied in mass mailing.  the dropable zone is also removed, causing a traceback error when attempting to add a new block without the dropable zone present.

To solve this issue, instead of assigning the innerHTML of the editable, I have assign the innerHTML of the droppable zone to the codeview. This ensures that the dropable zone is not removed when all content in the codeview is cleared.

Specifically, "this.wysiwyg.$editable.find('.o_editable')[0].innerHTML" will give us the innerHTML of the dropable zone.

Reply:
Ok I understand the problem now, thanks. Do you think it would be possible to do this change in the html field of mass mailing only ? And also use `querySelector` rather than jQuery `find`.

Reply:
@dmo-odoo Done the suggested changes.
I have checked both versions 15.0 and 16.0 and found out that in version 16.0, there is a file named "mass_mailing_html_field" which is used to toggle the codeview. However, I was not able to find anything like this in version 15.0. In 15.0 the _toggleCodeView method is used to toggle codeview, and when I tried the same code in same file in version 16.0, it did not work. To fix this, I need to do the same changes done here in the "mass_mailing_html_field" file.

Now, the question is whether I need to create a new branch for version 16.0. I don't think forward porting will work because the "field_html" and "_toggleCodeView" are still available in version 16.0.

Reply:
The mass mailing html field in 15.0 is at addons/mass_mailing/static/src/js/mass_mailing_widget.js. Does it help ?

Reply:
@dmo-odoo yes, it's working. Thanks for the suggestion.

Reply:
@dhba-odoo Thank you for the changes. Unfortunately, it looks like my advice to only do it in the html field of mass mailing was not a good one. In particular, in your recent changes you do not call super anymore, which means if we ever make some fix in the standard html field, we will not have the fix in mass mailing for example. It made me think if I could find a better place to do your fix and I think I have an idea.

- In the standard field html, replace `$codeview.val(this.$content.html());` by `$codeview.val(this.wysiwyg.getValue());`.
- Nothing changes in mass mailing field html (mass_mailing_widget) compared to 15.0, which means removing your current changes (the logic will be moved to wysiwyg, see next point).
- In the wysiwyg of mass_mailing, you can override `getValue` and `setValue` to check if the value contains the droppable zone and, if it does, use the editable instead, like you already do.

Let me know if it works, thanks !

Reply:
@dmo-odoo It's working fine overall, but one of the test tours failed during the process.

Reply:
@dmo-odoo `test_01_mass_mailing_editor_tour` This test tour failed here. the problem is generated from the following content in a test tour
**'verify that the save failed (since the field "subject" was not set and it is required)'**
when this content is run somehow it removes o_editable from dom and because of that in the next step of the tour o_editable is not found and the test tour failed.
I have tried doing the same manually and it's working. Also when I remove that specific step from the test tour at that time the test tour runs successfully.

What can be done here?